packages/coreutils/src/settingregistry.ts

```diff
@@ -354,6 +357,16 @@ export class SettingRegistry implements ISettingRegistry {
     const plugins = this.plugins;
     const registry = this;
 
+    if (fetched === undefined) {
+      throw [
+        {
+          dataPath: '',
+          keyword: 'id',
+          message: `Could not fetch settings for ${plugin}.`,
+          schemaPath: ''
+        } as ISchemaValidator.IError
+      ];
+    }
```

Comment:
@afshin Is this a reasonable error to throw?

Reply:
If you want to throw an error here, you need to put the `const fetched ...` line in a `try` block, because it can `throw`.

Reply:
I'm not sure if I understand the reasoning here. Could you spell it out for me? :)

Reply:
I don't think there is a circumstance where `fetched === undefined`. But the `const fetched = await ...` line can `throw` an error and if you are trying to intercept all `fetch` failures to format the error message as an `ISchemaValidator.IError`, I think you need to put the fetch call into a try/catch block.

This might also be a good time to change `ISchemaValidator.IError` from an `interface` to a `type`: `ISchemaValidator.Error`, but I'm not sure.

Reply:
To be more explicit: This code is trying to handle the case where `fetched` resolves as `undefined`, which is supposed to happen if the `id` (here `plugin`) is non-existent.

Reply:
(note I based the "resolves as undefined" on both the docstring and typings...

Reply:
Oh! Hm, is that what is supposed to happen? Then I withdraw my comment!

Reply:
ok, resolving this code as acceptable then :)